pkg/operator/controller/nodeobservability/daemonset.go

```diff
@@ -30,15 +30,15 @@ const (
 // ensureDaemonSet ensures that the daemonset exists
 // Returns a Boolean value indicating whether it exists, a pointer to the
 // daemonset and an error when relevant
-func (r *NodeObservabilityReconciler) ensureDaemonSet(ctx context.Context, nodeObs *v1alpha1.NodeObservability, sa *corev1.ServiceAccount) (bool, *appsv1.DaemonSet, error) {
-	nameSpace := types.NamespacedName{Namespace: nodeObs.Namespace, Name: daemonSetName}
-	desired := r.desiredDaemonSet(nodeObs, sa)
+func (r *NodeObservabilityReconciler) ensureDaemonSet(ctx context.Context, nodeObs *v1alpha1.NodeObservability, sa *corev1.ServiceAccount, ns string) (bool, *appsv1.DaemonSet, error) {
+	nameSpace := types.NamespacedName{Namespace: ns, Name: daemonSetName}
+	desired := r.desiredDaemonSet(nodeObs, sa, ns)
 	exist, current, err := r.currentDaemonSet(ctx, nameSpace)
 	if err != nil {
 		return false, nil, fmt.Errorf("failed to get DaemonSet: %w", err)
 	}
 	if !exist {
-		cmExists, err := r.createConfigMap(ctx, nodeObs)
+		cmExists, err := r.createConfigMap(ctx, nodeObs, ns)
```

Comment:
Preferably, `ensureDaemonSet()` should only ensure the `DaemonSet` and not ensure its dependencies (configmap) within it. Better to flatten out `Reconcile()` than to introduce nested dependency hierarchy. WDYT?

Reply:
Thanks for taking a look. I don't wanna touch code that's outside of scope of this PR.
We should do a refactor work once we reach desired functionality.